src/bucket/test/BucketListTests.cpp

```diff
@@ -337,58 +386,118 @@ TEST_CASE_VERSIONS("bucket list shadowing pre/post proto 12",
+        };
+
+        auto countNonBottomLevelEntries = [&] {
+            auto size = 0;
+            for (uint32_t i = 0; i < HotArchiveBucketList::kNumLevels - 1; ++i)
+            {
+                auto& level = bl.getLevel(i);
+                size += countEntries(level.getCurr());
+                size += countEntries(level.getSnap());
+            }
+            return size;
+        };
+
+        // Populate a BucketList so everything but the bottom level is full.
+        UnorderedSet<LedgerKey> keys;
+        auto numExpectedEntries = 0;
+        auto ledger = 1;
+        while (lastSnapSize() == 0)
+        {
+            bl.addBatch(*app, ledger, getAppLedgerVersion(app), {},
+                        LedgerTestUtils::generateValidUniqueLedgerKeysWithTypes(
+                            {CONTRACT_CODE, CONTRACT_DATA}, 5, keys),
+                        LedgerTestUtils::generateValidUniqueLedgerKeysWithTypes(
+                            {CONTRACT_CODE, CONTRACT_DATA}, 5, keys));
+
+            // Once all entries merge to the bottom level, only deleted entries
+            // should remain
+            numExpectedEntries += 5;
+
+            ++ledger;
+        }
+
+        // Close ledgers until all entries have merged into the bottom level
+        // bucket
+        while (countNonBottomLevelEntries() != 0)
+        {
+            bl.addBatch(*app, ledger, getAppLedgerVersion(app), {}, {}, {});
+            ++ledger;
+        }
+
+        auto bottomCurr =
+            bl.getLevel(HotArchiveBucketList::kNumLevels - 1).getCurr();
+        REQUIRE(countEntries(bottomCurr) == numExpectedEntries);
+
+        for (HotArchiveBucketInputIterator iter(bottomCurr); iter; ++iter)
+        {
+            auto be = *iter;
+            REQUIRE(be.type() == HOT_ARCHIVE_DELETED);
+            REQUIRE(keys.find(be.key()) != keys.end());
+        }
+    });
+}
+
+TEST_CASE_VERSIONS("live bucket tombstones expire at bottom level",
                    "[bucket][bucketlist][tombstones]")
 {
     VirtualClock clock;
     Config const& cfg = getTestConfig();
 
     for_versions_with_differing_bucket_logic(cfg, [&](Config const& cfg) {
         Application::pointer app = createTestApplication(clock, cfg);
-        BucketList bl;
+        LiveBucketList bl;
         BucketManager& bm = app->getBucketManager();
         auto& mergeTimer = bm.getMergeTimer();
         CLOG_INFO(Bucket, "Establishing random bucketlist");
-        for (uint32_t i = 0; i < BucketList::kNumLevels; ++i)
+        for (uint32_t i = 0; i < LiveBucketList::kNumLevels; ++i)
         {
             auto& level = bl.getLevel(i);
-            level.setCurr(Bucket::fresh(
+            level.setCurr(LiveBucket::fresh(
                 bm, getAppLedgerVersion(app), {},
-                LedgerTestUtils::generateValidUniqueLedgerEntriesWithExclusions(
-                    {CONFIG_SETTING}, 8),
+                LedgerTestUtils::generateValidUniqueLedgerEntries(8),
```

Comment:
We can't have `CONFIG_SETTINGS` types as a `deadEntry` or `initEntry` due to invariants, since `CONFIG_SETTING` can't be deleted and is guarenteed to already exist, so it can't be initialized either. However, it can (and does) get updated periodically, so I adjusted the test accordingly for slightly better coverage.